### DISCUSSION TITLE
font rendering can be improved

### DISCUSSION THREAD

**Discussion by rajputrajat** [wezterm/wezterm#389] [Q&A]
I tried my best to play with various configuration options, still I find font rendering in Alacritty better and more readable.
here is my conf:
```
local wezterm = require 'wezterm'
return {
    font = wezterm.font("Hack NF"),
    font_size = 9.0,
    line_height = 1.4,
    font_antialias = "Subpixel",
    dpi = 96.0,
    font_hinting = "Full",
    bold_brightens_ansi_colors = true,
    color_scheme = "Ayu",
    scrollback_lines = 10000,
    default_prog = {"C:\\Program Files\\PowerShell\\7\\pwsh.exe"},
}
```
[
![wez](https://user-images.githubusercontent.com/1040077/102801858-cf50a600-43db-11eb-99e6-eb98cc7b2194.png)
![ala](https://user-images.githubusercontent.com/1040077/102801864-d11a6980-43db-11eb-9b34-fa1159fa0d28.png)

](url)

**Comment by wez** [comment]
Could you try to put in words specifically what it is about alacritty's rendering that you prefer?  It's difficult to come back with a constructive response for you because there aren't any points to to respond to.

You may wish to try master; I just pushed 979368c4eb7f00dcae63d6336a3f3ef112f44f73 which adds a new `freetype_interpreter_version` option.  Previously, wezterm master forced the freetype interpreter version to 38.  That commit restores the version used in the last released version of wezterm, which is to use the freetype library default version (currently 40), and it allows you to select a version (current valid versions are 35, 38 and 40).  https://freetype.org/freetype2/docs/subpixel-hinting.html#subpixel-hinting has more information about the major differences.

**Reply by rajputrajat** [reply]
I apologize for not making my point clearer and thanks for your work.
I am using a 23" 1080p dell monitor for my work and have been using Alacritty as terminal on windows. Not having any term-mux always have been a pain.
Finding Wezterm is a ray of hope for me.
Now to the problem, when I look at text in Wezterm, It seem seem slightly hazy to me.
In comparison, the same text in Alacritty looks better, has more contrast to it. Alacritty text still is not perfect for me. but that maybe because low dpi screen and the font I am using.
I shared zoomed in screenshots from both for this purpose only.
I will try `freetype_interpreter_version` option and let you know, if that helps.

**Comment by quark-zju** [comment]
This seems like a sub-pixel smoothing difference.

**Comment by m2x07** [comment]
I had a similar problem. i found Alacritty's fonts to be more crisp. I played around with configuration options and found few settings that made fonts more "crisp" or readable as i should say. These are the options i added:

```lua
freetype_interpreter_version = 35        -- didn't seem to do much
freetype_load_target = "Light"    -- as "Light" is default hinting on my desktop (Gnome)
freetype_render_target = "HorizontalLcd" -- setting this to "light" had little to no visible effect  
```

i do not fully understand what these options do, but i think you might find this useful. my screen's specifications are 15-16inch approx, 1920x1080 @ 144hz, 96x96 resolution (from `xdpyinfo | grep resolution`)
